modules/objdetect/src/qrcode.cpp

```diff
@@ -976,23 +986,85 @@ class QRDecode
     size_t getVersion() { return version; }
     std::string getDecodeInformation() { return result_info; }
     bool fullDecodingProcess();
+    bool curvedDecodingProcess();
 protected:
     bool updatePerspective();
     bool versionDefinition();
     bool samplingForVersion();
     bool decodingProcess();
-    Mat original, no_border_intermediate, intermediate, straight;
+    inline double pointPosition(Point2f a, Point2f b , Point2f c);
+    float distancePointToLine(Point2f a, Point2f b , Point2f c);
+    bool getPointsInsideQRCode(vector<Point2f> angle_list);
+    bool computeClosestPoints(vector<Point> result_integer_hull);
+    bool computeSidesPoints(vector<Point> result_integer_hull);
+    vector<Point> getPointsNearUnstablePoint(vector<Point>&side, int start, int end, int step);
+    bool findAndAddStablePoint(vector<Point> result_integer_hull);
+    bool findIndexesCurvedSides();
+    bool findIncompleteIndexesCurvedSides();
+    Mat getMarkersMask();
```

Comment:
for more informatively 
```suggestion
    Mat getPatternsMask();
```